front/tests/012-op-simulation-settings-tab.spec.ts

```diff
@@ -0,0 +1,363 @@
+
+    await opInputTablePage.fillTableCellByStationAndHeader(
+      cell.stationName,
+      translatedHeader,
+      cell.value,
+      selectedLanguage
+    );
+    // Activate electrical profiles
+    await scenarioPage.openTabByDataId('tab-simulation-settings');
+    await opSimulationSettingsPage.activateElectricalProfile();
+    // Add the train schedule and verify output results
+    await scenarioPage.addTrainSchedule();
+    await scenarioPage.returnSimulationResult();
+    // TODO: Remove the reload when bug #8854 (UI not updating after modification) is fixed
+    await page.reload();
+    await opTimetablePage.verifyTimeStopsDataSheetVisibility();
+    await opTimetablePage.getTrainArrivalTime('12:06');
+    await opTimetablePage.clickOnScenarioCollapseButton();
+    await scrollContainer(page, '.osrd-simulation-container .time-stops-datasheet .dsg-container');
+    await opOutputTablePage.getOutputTableData(
+      expectedCellDataElectricalProfileON,
+      selectedLanguage
+    );
+    await opTimetablePage.clickOnTimetableCollapseButton();
+    // Deactivate electrical profiles and verify output results
+    await opTimetablePage.clickOnEditTrain();
+    await scenarioPage.openTabByDataId('tab-simulation-settings');
+    await opSimulationSettingsPage.deactivateElectricalProfile();
+    await opTimetablePage.clickOnEditTrainSchedule();
+    // TODO: Remove the reload when bug #8854 (UI not updating after modification) is fixed
+    await page.reload();
+    await opTimetablePage.verifyTimeStopsDataSheetVisibility();
+    await opTimetablePage.getTrainArrivalTime('12:05');
+    await opTimetablePage.clickOnScenarioCollapseButton();
+    await opOutputTablePage.getOutputTableData(
+      expectedCellDataElectricalProfileOFF,
+      selectedLanguage
+    );
+  });
+  test('should correctly activate composition code', async ({ page }) => {
+    const [
+      opInputTablePage,
+      opTimetablePage,
+      opOutputTablePage,
+      scenarioPage,
+      opSimulationSettingsPage,
+    ] = [
+      new OperationalStudiesInputTablePage(page),
+      new OperationalStudiesTimetablePage(page),
+      new OperationalStudiesOutputTablePage(page),
+      new ScenarioPage(page),
+      new OperationalStudiesSimulationSettingsPage(page),
+    ];
+
+    const translations = selectedLanguage === 'English' ? enTranslations : frTranslations;
+    const cell: CellData = {
+      stationName: 'Mid_East_station',
+      header: 'stopTime',
+      value: '124',
+    };
+    const translatedHeader = cleanWhitespace(translations[cell.header]);
+
+    await opInputTablePage.fillTableCellByStationAndHeader(
+      cell.stationName,
+      translatedHeader,
+      cell.value,
+      selectedLanguage
+    );
+    // Select a specific composition code option
+    await scenarioPage.openTabByDataId('tab-simulation-settings');
+    await opSimulationSettingsPage.deactivateElectricalProfile();
+    await opSimulationSettingsPage.activateMarecoMargin();
```

Comment:
Same than electrical profile switch, shouldn't this be activated by default ? Should we rather check that it is ?

Reply:
This one deactive the elecrical profile

Reply:
I meant the linear/mareco switch should be on "mareco" by default so we shouldn't have to activate it but rather check that it's on "mareco" no ?

Reply:
Added a new function to check "mareco"